Cython/Utility/MemoryView_C.c

```diff
@@ -26,8 +26,59 @@ typedef struct {
 #endif
 
 #define __pyx_atomic_int_type int
+#define __pyx_nonatomic_int_type int
+
+// for standard C/C++ atomics, get the headers first so we have ATOMIC_INT_LOCK_FREE
+// defined when we decide to use them
+#if CYTHON_ATOMICS && (defined(__STDC_VERSION__) && \
+                        (__STDC_VERSION__ >= 201112L) && \
+                        !defined(__STDC_NO_ATOMICS__))
+    #include <stdatomic.h>
+#elif CYTHON_ATOMICS && (defined(__cplusplus) && ( \
+                    (__cplusplus >= 201103L) || \
+                    (defined(_MSC_VER) && _MSC_VER >= 1700)))
+    #include <atomic>
+#endif
 
-#if CYTHON_ATOMICS && (__GNUC__ >= 5 || (__GNUC__ == 4 && \
+#if CYTHON_ATOMICS && (defined(__STDC_VERSION__) && \
+                        (__STDC_VERSION__ >= 201112L) && \
+                        !defined(__STDC_NO_ATOMICS__) && \
+                       ATOMIC_INT_LOCK_FREE == 2)
+    /* C11 atomics are available
+       require ATOMIC_INT_LOCK_FREE because I'm nervous about the __pyx_atomic_int[2]
+       alignment trick in MemoryView.pyx if it uses mutexes
+     */
```

Comment:
I don't see much value in leaving these in the generated C files, but I'll leave the decision to you.
However, I do prefer honouring the readers of my comments by presenting them complete sentences and correct punctuation wherever it seems appropriate. I feel that this applies here. :)
```suggestion
    // C11 atomics are available.
    // Require ATOMIC_INT_LOCK_FREE because I'm nervous about the __pyx_atomic_int[2]
    // alignment trick in MemoryView.pyx if it uses mutexes.
```

Reply:
I'm not sure it was a deliberate decision to leave these in the C files.